### PR TITLE
Allow `requests.post` to accept `Dict[str, str]` for `data`.

### PR DESCRIPTION
**Pull request by rowillia**
For example:

```python
import requests
requests.post('https://www.foo.bar', data={'hello': 'world'})
```

Failes to type check in Python 2 even though it's fine.

### CODE REVIEW

File: third_party/2and3/requests/api.pyi [hunk 1]
```diff
@@ -4,23 +4,28 @@ from typing import Optional, Union, Any, Iterable, Mapping, MutableMapping, Tupl
...
 
-_ParamsMappingValueType = Union[Text, bytes, int, float, Iterable[Union[Text, bytes, int, float]]]
-_Data = Union[None, bytes, MutableMapping[Text, Text], Iterable[Tuple[Text, Text]], IO]
+if sys.version_info >= (3,):
+    _Text = str
+else:
+    _Text = Union[str, Text]
+
+_ParamsMappingValueType = Union[_Text, bytes, int, float, Iterable[Union[_Text, bytes, int, float]]]
+_Data = Union[None, bytes, MutableMapping[_Text, _Text], Iterable[Tuple[_Text, _Text]], IO]
```
Comment (JelleZijlstra) [inline third_party/2and3/requests/api.pyi]: Because of invariance, won't this still get `{'a': 'b'}` rejected in Python 2?

**Comment by rowillia** [comment]
@JelleZijlstra you're right.  Does this mean I have to explode the possible combinations like this to get the desired behavior?

**Comment by JelleZijlstra** [comment]
Unfortunately, yes (and this still won't allow dictionaries that have both str and unicode keys).

For Iterable it's not necessary though, because that is a covariant class. If you can get away with Mapping instead of MutableMapping (I haven't checked), then the value, but not the key, type is also covariant.

**Comment by rowillia** [comment]
Got it.  Thanks @JelleZijlstra , updated this PR.

**Pull request merged**